FileUploadApi.md

```diff
@@ -0,0 +1,263 @@
+   "status":"IN_PROGRESS",
+   "fileDefinitions":[  
+      {  
+         "id":"88dfac11-1caf-4470-9ad1-d533f6360bdd",
+         "uploadDefinitionId":"67dfac11-1caf-4470-9ad1-d533f6360bdd",
+         "jobExecutionId":"66dfac11-1caf-4470-9ad1-d533f6360bdd",
+         "loaded":true,
+         "name":"marc.mrc"
+      },
+      {  
+         "id":"77dfac11-1caf-4470-9ad1-d533f6360bdd",
+         "uploadDefinitionId":"67dfac11-1caf-4470-9ad1-d533f6360bdd",
+         "jobExecutionId":"55dfac11-1caf-4470-9ad1-d533f6360bdd",
+         "loaded":false,
+         "name":"marc2.mrc"
+      }
+   ]
+}
+```
+
+##### Upload last file of the UploadDefinition
+
+```
+curl -w '\n' -X POST -D -   \
+   -H "Content-type: application/octet-stream"   \
+   -d @marc2.mrc \
+   http://localhost:9130/data-import/upload/file?uploadDefinitionId=67dfac11-1caf-4470-9ad1-d533f6360bdd&fileId=77dfac11-1caf-4470-9ad1-d533f6360bdd
+```
+
+##### Response with changed UploadDefinition
+
+As far as last file will be loaded, UploadDefinition change status to "LOADED"
+```
+{  
+   "id":"67dfac11-1caf-4470-9ad1-d533f6360bdd",
+   "metaJobExecutionId":"99dfac11-1caf-4470-9ad1-d533f6360bdd",
+   "status":"LOADED",
+   "fileDefinitions":[  
+      {  
+         "id":"88dfac11-1caf-4470-9ad1-d533f6360bdd",
+         "uploadDefinitionId":"67dfac11-1caf-4470-9ad1-d533f6360bdd",
+         "jobExecutionId":"66dfac11-1caf-4470-9ad1-d533f6360bdd",
+         "loaded":true,
+         "name":"marc.mrc"
+      },
+      {  
+         "id":"77dfac11-1caf-4470-9ad1-d533f6360bdd",
+         "uploadDefinitionId":"67dfac11-1caf-4470-9ad1-d533f6360bdd",
+         "jobExecutionId":"55dfac11-1caf-4470-9ad1-d533f6360bdd",
+         "loaded":true,
+         "name":"marc2.mrc"
+      }
+   ]
+}
+```
+
+### Delete FileDefinition
+
+##### Request for deleting second file
+
+```
+curl -X DELETE -D - -w '\n' http://localhost:9130/data-import/upload/definition/file/77dfac11-1caf-4470-9ad1-d533f6360bdd?uploadDefinitionId=67dfac11-1caf-4470-9ad1-d533f6360bdd
+```
+
+##### Response for deleting second file will be 204
+
+UploadDefinition entity will be changed to following state
+
+```
+{  
+   "id":"67dfac11-1caf-4470-9ad1-d533f6360bdd",
+   "metaJobExecutionId":"99dfac11-1caf-4470-9ad1-d533f6360bdd",
+   "status":"LOADED",
+   "fileDefinitions":[  
+      {  
+         "id":"88dfac11-1caf-4470-9ad1-d533f6360bdd",
+         "uploadDefinitionId":"67dfac11-1caf-4470-9ad1-d533f6360bdd",
+         "jobExecutionId":"66dfac11-1caf-4470-9ad1-d533f6360bdd",
+         "loaded":true,
+         "name":"marc.mrc"
+      }
+   ]
+}
+```
+
+### Add new FileDefinition to the existing UploadDefinition
+
+##### Request for adding new file
+
+```
+curl -w '\n' -X POST -D -   \
+   -H "Content-type: application/json"   \
+   -d @fileDefinition.json \
+   http://localhost:9130/data-import/upload/definition/file
+```
+
+Body of new fileDefinition should have uploadDefinitionId and file name.
```

Comment:
should has

Reply:
I checked with Grammarly that `should have` is correct.